helix-core/src/main/java/org/apache/helix/messaging/handling/AsyncCallbackService.java

```diff
@@ -62,15 +62,6 @@ void verifyMessage(Message message) {
       throw new HelixException(errorMsg);
     }
 
```

Comment:
Alternatively, if the exception is not graceful, we can return null when createHandler() is triggered. This is much better than returning a handler that does nothing (but still consume system resources).
Of course, the callers of createHandler() need to be improved to tolerate the null return value.

Reply:
You brought up a very good point, although for our scenario, ZK read probably fails before an effective DOS. I understand this reminds you of some TCP DOS. :) 

If you could check out `HelixTaskExecutor`, at around line 900, you can see how we handle null return value for `createHandler()`. The current approach is to skip the message without cleaning it up. I could explore on changing this current approach (because we need to clean up those reply messages), but such a change will impact other message handlings as well (such as state transition messages).